gp-includes/schema.php

```diff
@@ -58,15 +65,32 @@ function gp_schema_get() {
 	 * For example each WordPress Spanish translation (formal, informal and that of Diego) will be different sets.
 	 * Most projects will have only one translation set per locale.
 	 */
+
+	/*
+ 	 * The maximum length for the slug component of the project_id_slug_locale key is limited by the index size limit
+	 * minus the size of the project_id (4 bytes = 1 character) and locale (10 characters).
+	 *
+	 * Also make sure to never go over the length of the column.
+	 */
+	$max_pid_slug_locale_key_length = min( $max_index_length - 1 - 10, 255 );
```

Comment:
This is a little bit of future proofing, I played with a version of the PR that detected which database type was being used (MyISAM, InnoDB, etc.) and allowed for longer key lengths (InnoDB can support up to 3000+ as a key size) which could cause the length of a component of a key to be longer than the column itself.